### ISSUE TITLE
Mamba hanging on cloud BM and CI machines

### ISSUE THREAD

**Issue by esmalTT** [tenstorrent/tt-metal#12467]
### Summary

Multiple Mamba nightly tests are hanging after the following commit: 06b5b905ef0b6b3d3ca74f38118a20f57c3235b8 - as determined by a `bisect`. 

The hang triggers on a `.cpu()` call while setting the hidden state caches:

https://github.com/tenstorrent/tt-metal/blob/152902a59a042381ec9dc1468500ee22361fbc3c/models/demos/wormhole/mamba/tt/mamba_block.py#L201

But the stepping through the code using a debugger shows that it corrupted by an upstream `ttnn.to_layout` call:

https://github.com/tenstorrent/tt-metal/blob/152902a59a042381ec9dc1468500ee22361fbc3c/models/demos/wormhole/mamba/tt/mamba_block.py#L186

The input to this function is as follows:

```python
ttnn.Tensor([[[[-0.96094,  0.08008,  ...,  0.75000,  0.99219],
               [ 0.27344,  1.03125,  ..., -0.45703, -0.25195],
               ...,
               [-0.71484,  0.48438,  ..., -0.64844, -1.35156],
               [-0.77734,  0.37109,  ...,  3.87500,  0.21387]]]], shape=Shape([1, 1, 128, 5120]), dtype=DataType::BFLOAT16, layout=Layout::TILE)

MemoryConfig(memory_layout=TensorMemoryLayout::INTERLEAVED,buffer_type=BufferType::L1,shard_spec=std::nullopt)

```

### To Reproduce

Checkout 06b5b905ef0b6b3d3ca74f38118a20f57c3235b8 on a cloud BM machine and run the following:

```sh
pytest models/demos/wormhole/mamba/tests/test_mamba_model.py::test_inference -k "PREFILL and 128"
```

This bug does not show up on the lab machines I tested. Only in CI or on cloud BM machines.

To confirm this is fixed - Mamba should pass nightly pipelines on the following branch:

- https://github.com/tenstorrent/tt-metal/pull/12129

**Comment by tarafdarTT** [comment]
Sorry guys, should have it done today

**Comment by esmalTT** [comment]
Resolved in 0d20c99c3c8e86226e6ac70f09f5d4577cc0a25a
